internal/credentials/kubernetes/gar/service_account_key.go

```diff
@@ -37,7 +38,13 @@ func NewServiceAccountKeyProvider() credentials.Provider {
 }
 
 func (p *ServiceAccountKeyProvider) Supports(credType credentials.Type, repoURL string, data map[string][]byte) bool {
-	if credType != credentials.TypeImage || data == nil || data[serviceAccountKeyKey] == nil {
+	if (credType != credentials.TypeImage && credType != credentials.TypeHelm) ||
```

Comment:
Nit: This feels easier to comprehend, although this isn't a hill I'd die on. Up to you.

```suggestion
	if !(credType == credentials.TypeImage || credType == credentials.TypeHelm) ||
```

Reply:
de morgan's law prevents this from happening